pkg/pipelinedb/db.go

```diff
@@ -43,11 +43,42 @@ func NewDB(ctx context.Context, path string) (*DB, error) {
 	return OpenDB(ctx, path, false)
 }
 
-func OpenDB(ctx context.Context, path string, readOnly bool) (_ *DB, err error) {
-	if _, err := os.Stat(path); err != nil {
-		return nil, err
+func NewInMemoryDB(ctx context.Context) (*DB, error) {
```

Comment:
nit: Maybe `OpenAndCreate` or something? Usually `New` implies that the func doesn't do anything besides internally configuring. However, this also does quite a few transactions. The func above probably needs updating as well.